splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -693,9 +693,12 @@ public static void SYSCS_PERFORM_MAJOR_COMPACTION_ON_SCHEMA(String schemaName) t
      */
     public static void SYSCS_PERFORM_MAJOR_COMPACTION_ON_TABLE(String schemaName,String tableName) throws SQLException {
         LanguageConnectionContext lcc = (LanguageConnectionContext) ContextService.getContext(LanguageConnectionContext.CONTEXT_ID);
+        assert lcc != null;
         DataDictionary dd = lcc.getDataDictionary();
         // sys query for table conglomerate for in schema
         PartitionFactory tableFactory = SIDriver.driver().getTableFactory();
+        schemaName = EngineUtils.validateSchema(schemaName);
+        tableName = EngineUtils.validateTable(tableName);
```

Comment:
IT failed due to this, SYSCS_PERFORM_MAJOR_COMPACTION_ON_SCHEMA calls this function with tableName set to null. We could perform the validate here and move all the logic into a separate function that gets called both by SYSCS_PERFORM_MAJOR_COMPACTION_ON_TABLE and SYSCS_PERFORM_MAJOR_COMPACTION_ON_SCHEMA

Reply:
Fixed!